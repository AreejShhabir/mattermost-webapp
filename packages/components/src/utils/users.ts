// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {UserProfile} from '@mattermost/types/users';

import Preferences from '../constants/preferences';

import {localizeMessage} from './i18n';

export function getFullName(user: UserProfile): string {
    if (user.first_name && user.last_name) {
        return user.first_name + ' ' + user.last_name;
    } else if (user.first_name) {
        return user.first_name;
    } else if (user.last_name) {
        return user.last_name;
    }

    return '';
}

export function displayUsername(
    user: UserProfile | null | undefined,
    teammateNameDisplay: string,
    useFallbackUsername = true,
): string {
    let name = useFallbackUsername ? localizeMessage('channel_loader.someone', 'Someone') : '';
    if (user) {
        if (teammateNameDisplay === Preferences.DISPLAY_PREFER_NICKNAME) {
            name = user.nickname || getFullName(user);
        } else if (teammateNameDisplay === Preferences.DISPLAY_PREFER_FULL_NAME) {
            name = getFullName(user);
        } else {
            name = user.username;
        }

        if (!name || name.trim().length === 0) {
            name = user.username;
        }
    }

    return name;
}
