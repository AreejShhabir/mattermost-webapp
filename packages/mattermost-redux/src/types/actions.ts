// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {ThunkAction} from 'redux-thunk';
import {AnyAction} from 'redux';
import {BatchAction} from 'redux-batched-actions';

import {GlobalState} from '@mattermost/types/store';
import {ServerError} from '@mattermost/types/errors';

export type GetStateFunc = () => GlobalState;
export type GenericAction = AnyAction;
export type Thunk = (b: DispatchFunc, a: GetStateFunc) => Promise<ActionResult> | ActionResult;

export type Action = GenericAction | Thunk | BatchAction | ActionFunc;

export type ActionResult<Data = unknown, Error = unknown> = {
    data?: Data;
    error?: Error;
};

export type DispatchFunc = (
    action: Action,
    getState?: GetStateFunc | null
) => Promise<ActionResult>;

export type ActionFunc<Data = any, State = GlobalState, Error = ServerError, ExtraThunkArg = null> = ThunkAction<
Promise<ActionResult<Data, Error>> | ActionResult<Data, Error>,
State,
ExtraThunkArg,
AnyAction
>;
