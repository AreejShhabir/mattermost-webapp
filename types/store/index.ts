// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Channel} from '@mattermost/types/channels';
import {GlobalState as BaseGlobalState} from '@mattermost/types/store';
import {RelationOneToOne} from '@mattermost/types/utilities';

import {NewPostDraft} from './draft';
import {PluginsState} from './plugins';
import {ViewsState} from './views';

export type DraggingState = {
    state?: string;
    type?: string;
    id?: string;
}

export type GlobalState = BaseGlobalState & {
    plugins: PluginsState;
    storage: {
        storage: Record<string, any>;
        initialized: boolean;
    };
    views: ViewsState;
    drafts: {
        byChannel: RelationOneToOne<Channel, NewPostDraft>;
        byThread: RelationOneToOne<Channel, NewPostDraft>;
    };
};
