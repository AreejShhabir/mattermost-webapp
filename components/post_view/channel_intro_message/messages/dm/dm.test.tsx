// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {UserProfile} from '@mattermost/types/users';
import {bot1, directChannel, user1} from '../test_utils';

import DMIntroMessage from './dm';

const component = (otherProps: any) => {
    return (
        <DMIntroMessage
            channel={directChannel}
            {...otherProps}
        />
    );
};

describe('components/post_view/ChannelIntroMessages', () => {
    describe('test DIRECT Channel', () => {
        test('should match snapshot, without teammate', () => {
            expect(shallow(component({}))).toMatchSnapshot();
        });

        test('should match snapshot, with teammate', () => {
            expect(shallow(component({
                teammate: user1 as UserProfile,
                teammateName: 'my teammate',
            }))).toMatchSnapshot();
        });

        test('should match snapshot, with bot teammate', () => {
            expect(shallow(component({
                teammate: bot1 as UserProfile,
                teammateName: 'my teammate',
            }))).toMatchSnapshot();
        });
    });
});
