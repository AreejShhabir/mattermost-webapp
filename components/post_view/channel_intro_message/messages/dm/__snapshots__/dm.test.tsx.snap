// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_view/ChannelIntroMessages test DIRECT Channel should match snapshot, with bot teammate 1`] = `
<Fragment>
  <div
    className="post-profile-img__container channel-intro-img"
  >
    <ProfilePicture
      hasMention={true}
      isEmoji={false}
      isRHS={false}
      popoverPlacement="right"
      size="xl"
      src="/api/v4/users/bot1/image?_=0"
      userId="bot1"
      wrapperClass=""
    />
  </div>
  <div
    className="channel-intro-profile d-flex"
  >
    <Connect(UserProfile)
      disablePopover={false}
      hasMention={true}
      userId="bot1"
    />
  </div>
  <p
    className="channel-intro-text"
  >
    <FormattedMarkdownMessage
      defaultMessage="This is the start of your direct message history with {teammate}.\\\\nDirect messages and files shared here are not shown to people outside this area."
      id="intro_messages.DM"
      values={
        Object {
          "teammate": "my teammate",
        }
      }
    />
  </p>
  <Connect(Component)
    show={false}
  />
  <Connect(Component)
    show={false}
  />
</Fragment>
`;

exports[`components/post_view/ChannelIntroMessages test DIRECT Channel should match snapshot, with teammate 1`] = `
<Fragment>
  <div
    className="post-profile-img__container channel-intro-img"
  >
    <ProfilePicture
      hasMention={true}
      isEmoji={false}
      isRHS={false}
      popoverPlacement="right"
      size="xl"
      src="/api/v4/users/user1/image?_=0"
      userId="user1"
      wrapperClass=""
    />
  </div>
  <div
    className="channel-intro-profile d-flex"
  >
    <Connect(UserProfile)
      disablePopover={false}
      hasMention={true}
      userId="user1"
    />
  </div>
  <p
    className="channel-intro-text"
  >
    <FormattedMarkdownMessage
      defaultMessage="This is the start of your direct message history with {teammate}.\\\\nDirect messages and files shared here are not shown to people outside this area."
      id="intro_messages.DM"
      values={
        Object {
          "teammate": "my teammate",
        }
      }
    />
  </p>
  <Connect(Component)
    show={true}
  />
  <Connect(Component)
    show={true}
  />
</Fragment>
`;

exports[`components/post_view/ChannelIntroMessages test DIRECT Channel should match snapshot, without teammate 1`] = `
<p
  className="channel-intro-text"
>
  <MemoizedFormattedMessage
    defaultMessage="This is the start of your direct message history with this teammate. Direct messages and files shared here are not shown to people outside this area."
    id="intro_messages.teammate"
  />
</p>
`;
